p2p/node/peerManager/peerManager.go

```diff
@@ -108,15 +109,8 @@ type PeerManager interface {
 	// RefreshBootpeers returns all the current bootpeers for bootstrapping
 	RefreshBootpeers() []peer.AddrInfo
 
-	// Increases the peer's liveliness score
-	MarkLivelyPeer(peerID p2p.PeerID, topic *pubsubManager.Topic)
-	// Decreases the peer's liveliness score
-	MarkLatentPeer(peerID p2p.PeerID, topic *pubsubManager.Topic)
-
-	// Increases the peer's liveliness score. Not exposed outside of NetworkingAPI
-	MarkResponsivePeer(peerID p2p.PeerID, topic *pubsubManager.Topic)
-	// Decreases the peer's liveliness score. Not exposed outside of NetworkingAPI
-	MarkUnresponsivePeer(peerID p2p.PeerID, topic *pubsubManager.Topic)
+	// Adjust the quality score of a peer by applying the given adjustment function
+	AdjustPeerQuality(p2p.PeerID, func(int) int)
```

Comment:
Is it better to ask the caller to specify the adjustment function as an argument, or to call a function based on a more descriptive name?

I don't have strong feelings but it seems like the adjustment functions would be harder to find compared to just looking at the peerManager interface.

Reply:
I prefer this sort of approach, because it makes it possible to modify/extend functionality without redefining the interface, which can have rippling consequences.

Either way would work though. I prefer this particular flavor, but if I'm the only one that feels that way, I can do it the other way.